docs/meta/roadmap.rst

```diff
@@ -0,0 +1,205 @@
+configuration management tools can be used to create these images with required
+configuration "baked-in" during the build process. In self-hosted or
+on-premises environments, configuration management can be used to configure
+hosts from scratch, including setting up hypervisors for virtual machines or
+container environments which then can leverage the container-based deployments.
+
+Configuration management best practices were developed over many years by
+experienced system administrators. The current set of CM tools can be used to
+easily capture that experience, share it with others and improve for the
+benefit of the entire community.
+
+
+Provide a set of building blocks for complex deployments
+--------------------------------------------------------
+
+Typical infrastructure deployments are composed from multiple services. There's
+the base operating system which defines the environment; a firewall service
+controls the access to other services; a webserver provides an access point to
+the static and dynamic applications; database services create different storage
+solutions available to the applications. Each of these services is usually
+managed as standalone, but they are meant to work together to achieve a task.
+
+The Ansible roles included in the DebOps project are meant to define a concise
+and stable API surface between different services. Each role should focus on
+specific service configuration and in turn, provide an interface for other
+roles where apropriate so that multiple services can be configured together
+without conflicts.
+
+This will allow the project to provide a set of "stacks" written as Ansible
+playbooks that use multiple Ansible roles together to deploy complex
+applications - a WorPress blog, a VM hypervisor host, a container cluster will
+be composed from the same set of building blocks, which allows creation of
+customized environments, tailored for a particular organization.
+
+
+Final goal: DebOps is used to set up a data center in an extraterrestrial colony
+--------------------------------------------------------------------------------
+
+Every project needs a final goal which, when reached, marks its completion. The
+final goal of the DebOps project is for it to be used on an extraterrestial
+colony to set up a data center.
+
+It's not a pipe dream. Debian has already made it into space, due to `being
+used as the operating system on various laptops used on the International
+Space Station`__. Humanity already made it to the Earth's Moon in the past, so
+we will be getting there at some point in the future; other planets like Mars
+will also be visited. Eventually a new human colony will be estabilished,
+either on a planetary/lunar surface, on an asteroid or within an artifical space
+station. In such case, at least one local data center will have to be set up to
+faciliate computing tasks optimally.
+
+Obviously the first couple of data centers set up in space won't be based on
```

Comment:
This sentence bugs me. I would like to be more optimistic here, I mean we are talking about the end goal. What else would you deploy first if not Free Software? What else would you trust enough to keep you alive? I only accept HAL 9000 or Free Software :) Oh wait, HAL 9000 did not keep us humans alive.

Reply:
Life support system, traffic control for local airspace, temperature control... These systems are better handled by a real time OS such as WxWorks. Will it be Free Software? Hopefully. But I don't think that will be Linux, unless RTOS support is improved.

We are talking about stuff done much later, meant for normal non-critical systems.

Reply:
Then we are on the same page. I was also thinking about life support systems, but the sentence allows for a different, more proprietary interpretation was well. I guess we can just be more specific.